### PR TITLE
push notifications service subscribe, unsubscribe and tests

### PR DESCRIPTION
**Pull request by gabrielbazan7**

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -2407,6 +2413,40 @@ WalletService.prototype.getFiatRate = function(opts, cb) {
...
+    url: config.pushNotificationsOpts.pushServerUrl + '/subscribe',
+    method: 'POST',
+    json: true,
+    body: opts
+  }, function(err, response) {
+    return cb(err, response);
+  });
+};
+
+WalletService.prototype.pushNotificationsUnsubscribe = function(opts, cb) {
```
Comment (isocolsky) [inline lib/server.js]: `opts` should always be an object.

File: lib/server.js [hunk 2]
```diff
@@ -2407,6 +2413,40 @@ WalletService.prototype.getFiatRate = function(opts, cb) {
...
+  var self = this;
+
+  if (opts) opts = {
+    token: opts
```
Comment (isocolsky) [inline lib/server.js]: This is not secure as one user could unsubscribe any other user.

**Comment by gabrielbazan7** [comment]
Updated

**Comment by matiu** [comment]
needs rebase

**Comment by gabrielbazan7** [comment]
Updated

File: lib/server.js [hunk 3]
```diff
@@ -2421,6 +2427,35 @@ WalletService.prototype.getFiatRate = function(opts, cb) {
   });
 };
 
+WalletService.prototype.pushNotificationsSubscribe = function(opts, cb) {
+  var self = this;
+
```
Comment (matiu) [inline lib/server.js]: Please add: 


  if (!Utils.checkRequired(opts, ['token']))
    return cb(new ClientError('Required argument missing'));

**Pull request merged**